andela/indiana-ah-backend
# #163426533 Implement email and in app notification

#### What does this PR do?
* Implement email and In-app notification when certain events happen
#### Description of Task to be completed?
Have the following endpoints working:

* Enable In-app notification:  PATCH `localhost:3000/api/v1/notifications/inApp`
* Opt out of  In-app notification:  PATCH `localhost:3000/api/v1/notifications/inApp`
* Enable email notification:  PATCH `localhost:3000/api/v1/notifications/email`
* Opt out of email notification:  PATCH `localhost:3000/api/v1/notifications/email`

Implemented the following functionalities
* Enable notification when a bookmarked article has a new reaction
* Enable notification when a user has a new follower
* Enable notification when a user you follow publishes an article 

#### How should this be manually tested?
* clone the repo
* run `npm install` to install new packages (Pusher)
* Make sure at least two users are created, a user has an article in the database, the user has at least a follower and has bookmarked an article
* fire up postman
* navigate to PATCH `localhost:3000/api/v1/notifications/inApp` to opt in for In-app notification
* navigate to PATCH `localhost:3000/api/v1/notifications/email` to opt in for email notification
* navigate to POST `http://localhost:3000/api/v1/profiles/<name of the user to follow>/follow` to follow  the user, the user will receive an email and In-app notification

#### What are the relevant pivotal tracker stories?
[#163426533](https://www.pivotaltracker.com/story/show/163426533)
#### Screenshots (if appropriate)


![image](https://user-images.githubusercontent.com/22154654/53561813-18bd6a80-3b50-11e9-9d0f-a17028f23723.png)
![image](https://user-images.githubusercontent.com/22154654/53561924-633ee700-3b50-11e9-91bf-3ac4f40ee9b4.png)
![image](https://user-images.githubusercontent.com/22154654/53561964-7782e400-3b50-11e9-997c-31e12b8471bb.png)
![image](https://user-images.githubusercontent.com/22154654/53790212-ba650300-3f26-11e9-8d68-e7d9d055782a.png)
![image](https://user-images.githubusercontent.com/22154654/53790327-fd26db00-3f26-11e9-86c2-f80d8325b655.png)
[image](https://user-images.githubusercontent.com/22154654/53790246-d799d180-3f26-11e9-969b-093ccc0ac194.png)
![image](https://user-images.githubusercontent.com/22154654/53790407-29425c00-3f27-11e9-9af6-307e2994a3fe.png)
